Rapid7-ThreatCommand/ThreatCommand_workflow.xml

```diff
@@ -0,0 +1,57 @@
+<Workflow name="Qmasters Rapid7 ThreatCommand workflow for QRadar" version="1.0" xmlns="http://qradar.ibm.com/UniversalCloudRESTAPI/Workflow/V2">
+    <Parameters>
+		<Parameter name="hostname" label="Host Name" required="true" />
+		<Parameter name="account_id" label="Account ID" required="true" secret="true" />
+		<Parameter name="api_key" label="API Key" required="true" secret="true" />
+    </Parameters>
+	
+    <Actions>
+		<!-- Clear the log source status before a new workflow run starts -->
+        <ClearStatus />
+		
+        <!-- Prepare the start date filter, based on the current bookmark -->
+		<Initialize path="/bookmark" value="${time() - 604800000}" />
+        
+        <!-- Prepare the end date filter, which is the current time -->
+        <Set path="/current_time" value="${time()}" />
+
+		<!-- Get Alert List -->				
+		<CallEndpoint url="${/hostname}/public/v2/data/alerts/alerts-list" method="GET" savePath="/get_alert_list" >
+			<BasicAuthentication username="${/account_id}" password="${/api_key}" />
+            <QueryParameter name="isClosed" value="false" />
+            <QueryParameter name="foundDateFrom" value="${/bookmark}" />
+            <QueryParameter name="foundDateTo" value="${/current_time}" />
+		</CallEndpoint>
+
+        <!-- Handle Errors -->
+        <If condition="/get_alert_list/status_code = 403">
+			<Abort reason="Invalid ThreatCommand API Key"/>
+		</If>
+        <If condition="/get_alert_list/status_code != 200">
+            <Abort reason="hreatCommand abort reason:  ${/get_alert_list}" />
+        </If>
+
+        <!-- Get Alerts -->	
+        <ForEach item="/alert_id" items="/get_alert_list/body">
+		    <CallEndpoint url="${/hostname}/public/v1/data/alerts/get-complete-alert/${/alert_id}" method="GET" savePath="/get_alerts/${/alert_id}" >
+		    	<BasicAuthentication username="${/account_id}" password="${/api_key}" />
+		    </CallEndpoint>
+
+            <!-- Post Events, if any -->
+            <If condition="count(/get_alerts) > 0">
+                <PostEvent path="/get_alerts/${/alert_id}/body" source="${/hostname}" />
+
+                <!-- Update the bookmark -->
+                <ParseDate pattern="yyyy-MM-dd'T'HH:mm:ss[.SSS]'Z'" timeZone="UTC" date="${/get_alerts/${/alert_id}/body/FoundDate}" savePath="/last_event_time" />
+                <FormatDate pattern="yyyy-MM-dd'T'HH:mm:ss.SSS'Z'" timeZone="UTC" time="${/last_event_time + 1}" savePath="/bookmark" />

```

Comment:
I'm not completely sure about how the was taken here into consideration.

How does the `FoundDate` work? Is it possible that you will get an old date after an early date?

are the alerts returned in `alert-list` sorted by time?
If so, we can change the logic here to take the last element within the list I created for you.
If not, we should re-consider how this logic works.